2312/2/rapper.rb

```diff
@@ -0,0 +1,61 @@
+require './word'
+require './lyrics'
+
+# this class provides all info about rapper
+class Rapper
+  attr_reader :stats, :name, :overall, :battles
+
+  def initialize(name, all_battles_paths)
+    @name = name
+    @battles = all_battles_paths.select { |battle_path| battle_path.split(/( против | vs )/i).first.include?(@name) }
+    @overall = { battles_num: @battles.size, rounds_num: Rapper.rapper_rounds(@battles) }
+    @stats = Stats.new(self)
+  end
+
+  def self.count_rounds(battle_file_path)
```

Comment:
Why don't you want to create Rapper object? Then we could you something like
```ruby
rapper = Rapper.new(args)
rapper.rounds_count
rapper.bad_words
```
And so on.